frontend/src/pages/StudentPage.tsx

```diff
@@ -1,18 +1,45 @@
-import { Typography } from '@mui/joy';
+import { ErrorOutline } from '@mui/icons-material';
+import { Snackbar, Typography } from '@mui/joy';
+import { useState } from 'react';
 import { AlternateButton } from '../components/AlternateButton.tsx';
 import { ContentSection } from '../components/ContentSection.tsx';
+import { CreateStudentModal } from '../components/CreateStudentModal.tsx';
 import { StudentTable } from '../components/StudentTable.tsx';
 
 export function StudentPage() {
+  const [open, setOpen] = useState(false);
+
+  const [showSnackbar, setShowSnackbar] = useState(false);
+  const [showSnackbarError, setShowSnackbarError] = useState(false);
   return (
     <ContentSection>
       <div style={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center' }}>
         <Typography level="h3" margin={0} lineHeight={1}>
           Schülerübersicht
         </Typography>
-        <AlternateButton>Schüler hinzufügen</AlternateButton>
+        <AlternateButton onClick={() => setOpen(!open)}>Schüler hinzufügen</AlternateButton>
```

Comment:
```suggestion
        <AlternateButton onClick={() => setOpen(true)}>Schüler hinzufügen</AlternateButton>
```
As we never close the modal over here.